plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetastoreAccessOperations.java

```diff
@@ -126,6 +133,54 @@ public void testSelectWithFilter()
                         .build());
     }
 
+    @Test
+    public void testSelectFromView()
+    {
+        assertUpdate("CREATE TABLE test_select_view_table (id VARCHAR, age INT)");
+        assertUpdate("CREATE VIEW test_select_view_view AS SELECT id, age FROM test_select_view_table");
+
+        assertMetastoreInvocations("SELECT * FROM test_select_view_view",
+                ImmutableMultiset.builder()
+                        .addCopies(GET_TABLE, 5)
+                        .build());
+    }
+
+    @Test
+    public void testSelectFromViewWithFilter()
+    {
+        assertUpdate("CREATE TABLE test_select_view_where_table AS SELECT 2 as age", 1);
+        assertUpdate("CREATE VIEW test_select_view_where_view AS SELECT age FROM test_select_view_where_table");
+
+        assertMetastoreInvocations("SELECT * FROM test_select_view_where_view WHERE age = 2",
+                ImmutableMultiset.builder()
+                        .addCopies(GET_TABLE, 5)
+                        .build());
+    }
+
+    @Test
+    public void testSelectFromMaterializedView()
```

Comment:
A case for refreshing the view might be useful?

Reply:
added